programs/dynamic-bonding-curve/src/instructions/partner/ix_create_damm_v2_config.rs

```diff
@@ -0,0 +1,271 @@
+use anchor_lang::{prelude::*, solana_program::clock::SECONDS_PER_DAY};
+use anchor_spl::token_interface::Mint;
+
+use crate::{
+    constants::{
+        MAX_LOCK_DURATION_IN_SECONDS, MAX_LOCK_DURATION_IN_SLOTS, MIN_LOCKED_LP_PERCENTAGE,
+    },
+    params::{
+        fee_parameters::PoolFeeParameters, liquidity_distribution::LiquidityDistributionParameters,
+    },
+    process_create_config,
+    safe_math::SafeMath,
+    state::{LpVestingInfo, MigrationFeeOption, MigrationOption},
+    utils_math::time_to_slot,
+    validate_common_config_parameters, CreateConfigCtx, EvtCreateDammV2Config, LockedVestingParams,
+    MigratedPoolFee, MigrationFee, PoolError, ProcessCreateConfigAccounts, ProcessCreateConfigArgs,
+    TokenSupplyParams, ValidateCommonConfigParametersArgs,
+};
+
+#[derive(AnchorSerialize, AnchorDeserialize, Debug, Clone)]
+pub struct LpDistributionInfo {
+    pub lp_percentage: u8,
+    pub lp_permanent_lock_percentage: u8,
+    pub lp_vesting_info: LpVestingInfoParams,
+}
+
+#[derive(AnchorSerialize, AnchorDeserialize, Debug, Clone, Copy)]
+pub struct LpVestingInfoParams {
+    pub vesting_percentage: u8,
+    pub cliff_duration_from_migration_time: u32,
+    pub bps_per_period: u16,
+    pub frequency: u64,
+    pub number_of_periods: u16,
+}
+
+impl LpVestingInfoParams {
+    pub fn validate(&self) -> Result<()> {
+        if self.to_lp_vesting_info().is_none() {
+            return Ok(());
+        }
+
+        require!(
+            self.vesting_percentage > 0 && self.vesting_percentage <= 100,
+            PoolError::InvalidVestingParameters
+        );
+
+        if self.number_of_periods > 0 {
+            require!(
+                self.frequency > 0 && self.bps_per_period > 0,
+                PoolError::InvalidVestingParameters
+            );
+        }
+
+        let total_bps_after_cliff =
+            u64::from(self.bps_per_period).safe_mul(self.number_of_periods.into())?;
+
+        require!(
+            total_bps_after_cliff <= 10_000,
+            PoolError::InvalidVestingParameters
+        );
+
+        // We not sure the damm v2 config will be in slot / time activation. Therefore, we validate both time + slot based to ensure that later during migration we won't have issue.
```

Comment:
we should update all damm v2 configs to timestamp, instead of supporting both, that make code harder to read

Reply:
Fixed